src/main/java/io/jenkins/plugins/checks/github/GitSCMChecksContext.java

```diff
@@ -49,7 +53,12 @@ class GitSCMChecksContext extends GitHubChecksContext {
     @Override
     public String getHeadSha() {
         try {
-            String head = getGitCommitEnvironment();
+            String head;
+            if (!this.config.getCommit().equals("")) {
+                head = getGitCommitEnvironment(this.config.getCommit());
+            } else {
+                head = getGitCommitEnvironment("GIT_COMMIT");
+            }
```

Comment:
This feels a bit more natural to me, but it might be even more even more natural to access the `config` directly inside `getGitCommitEnvironment()` and save passing a variable around.

```suggestion
            String commitEnvVar = this.config.getCommit();
            if (commitEnvVar.isEmpty()) {
                  commitEnvVar = "GIT_COMMIT";
            }
            String head = getGitCommitEnvironment(commitEnvVar);
```